LanguageFeatures/primary-constructors/grammar_A02_t03.dart

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Let `D` be a class, extension type, or enum declaration.
+///
+/// A compile-time error occurs if `D` includes a `<classNamePart>` that
+/// contains a `<primaryConstructorNoConst>`, and the body of `D` contains a
+/// `<declaringConstructorSignature>` that contains a `<declaringParameterList>`.
+///
+/// @description Check that it is a compile-time error if `D` includes a
+/// `<classNamePart>` that contains a `<primaryConstructorNoConst>`, and the
+/// body of `D` contains a `<declaringConstructorSignature>` that contains a
+/// `<declaringParameterList>`. Test an enum.
+/// @author sgrekhov22@gmail.com
+
+// TODO (sgrekhov) Add `declaring-constructors` experimental flag
+
+enum const E1(final int v) {
+  e1(1);
+
+  this();
+//    ^
+// [analyzer] unspecified
+// [cfe] unspecified
+}
+
+enum const E2() {
+  e2;
+
+  this();
+//    ^
+// [analyzer] unspecified
+// [cfe] unspecified
+}
+
+enum const E3(final int v) {
+  e3(3);
+
+  this(final int v);
+//    ^
+// [analyzer] unspecified
+// [cfe] unspecified
+}
+
+enum const E4(final int v) {
+  e4(4);
+
+  this(this.v);
+//    ^
+// [analyzer] unspecified
+// [cfe] unspecified
+}
+
+enum const E5.someName(final int v) {
+  e5.someName(5);
+
+  this.someName(final int v);
```

Comment:
Again, different names would also be an error.

Reply:
How about the same tests without the `const` modifier in the header? With an `enum` declaration, the constness of the constructor is inferred if it is not specified explicitly (yes, that's a weird exception, but that's what we got ;-).

Reply:
Added the case with different names. And the `const` modifier removed for enums with even numbers (I didn't know about this exception, thanks!).

Reply:
Sounds good!